features/custom_codecs/README.md

```diff
@@ -0,0 +1,379 @@
+## Custom Codecs
+
+[JAVA-721][JAVA-721] finally brings a frequently requested feature: to support out-of-the-box, transparent and user-configurable mapping of CQL types to arbitrary Java objects.
```

Comment:
Documentation is here to stay, so I think it should avoid time contextualization like "this was recently introduced", "this method is new", etc. It should be written in a way that will still be relevant in 10 versions if the feature is still there. For the same reasons, ticket number and "since" version are useful information, but we shouldn't lead with them.

Suggestions:

> [JAVA-721][JAVA-721] finally brings a frequently requested feature: to support out-of-the-box, transparent and user-configurable mapping of CQL types to arbitrary Java objects.

"Custom codecs allow transparent, user-configurable mapping of CQL types to arbitrary Java objects." (I also didn't understand "out-of-the-box" in this sentence, custom codecs are not out of the box since users provide them).

> The central piece in the new serialization mechanism introduced by JAVA-721 is [TypeCodec][TypeCodec], 
which is now part of the driver's public API.

"The central piece in the serialization mechanism is [TypeCodec]". No need to mention that it was internal before, that's not relevant to regular users who don't look at the source code, and those who hack the internals will figure it out themselves.

> `TypeCodec` is now intended to be freely subclassed by users willing to implement their own serialization logic.

"`TypeCodec` can be freely subclassed by users willing to implement their own serialization logic."

>     .set(1, myPojo, MyPojo.class); //newly-created set() method

Remove the comment.

> the new `get()` and `set()` methods

Remove "new". Maybe we could refer to these methods as "generic".

PS -- markdown tip: when the name and contents of the link are the same, you can omit the name. `[JAVA-721][JAVA-721]` becomes `[JAVA-721]`.